packages/clay-drop-down/src/hooks.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+
+const ESC_KEY_CODE = 27;
+
+/**
+ * Hook for closing dropdown when user hits ESC key or clicks outside the menu.
+ */
+export function useDropdownCloseInteractions(
+	elementRef: React.RefObject<HTMLElement>,
+	setActive: (val: boolean) => void
+) {
+	React.useEffect(() => {
+		const handleClick = (event: MouseEvent) => {
```

Comment:
It seems that when I click on the trigger to close the modal it closes and opens, for some reason it is losing its active state.

Reply:
good catch, i forgot to test that and I think I know the cause.